File: OLD Repos/Numerical Series and Squences/Sum/The Sums of Sequences.py
#### The Sums of Sequences

#  0. Time Interval Setting
import numpy as np
import time

#  1. Calculate e using Taylor Series

to = time.time()

s = 1.0
f = 1.0
for n in range(1,1001):
    f *= n
    s = s + 1/f
e = s

print("The natural logarithm is %f." %e)
td = time.time() - to
print("The time interval is %f s." %td)

#  2. Calculate π using Taylor Series

to = time.time()

s = 0.0
n = 0
for n in range(10000001):
    s = s + (-1)**n/(2*n+1)
pi = 4*s

print("The π is %f." %pi)
td = time.time() - to
print("The time interval is %f s." %td)

#  3. Calcute the sum(n!/n^n)

to = time.time()

s = 0
u = 1
d = 1
for n in np.arange(1,5,1):
    u = n * u           # n!
    d = n**n           # n^n
    s = s + u/d

print("The result of sum(n!/n^n) is %f" %s)
td = time.time() - to
print("The time interval is %f s." %td)

File: OLD Repos/Numerical Series and Squences/Sum/The Sums of Sequences.jl
#### The Sums of Sequences

#  1. Calculate e using Taylor Series

to = time()

e = 1.0
f = 1.0
for n in 1:1000
    global f = n * f
    global e = e + 1/f
end

println("The Natural logarithm is $e.")
td = time() - to
println("The time interval is $td s.")

#  2. Calculate π using Taylor Series

to = time()

s = 0.0
for n in 0:10000000
    global s = s + (-1)^n/(2n+1)
end
π = 4s

println("The π is ",π)
td = time() - to
println("The time interval is $td s.")

#  3. Calcute the sum(n!/n^n)

to = time()

s = 0.0
u = Rational(1)
d = Rational(1)
for n in 1:4
    global u = n * u        # n!
    global d = n ^ n        # n^n
    global s = s + u/d
end

println("The result of sum(n!/n^n) is ",s)
td = time() - to
println("The time interval is $td s.")